app/models/concerns/foreman_kernel_care/host_managed_extensions.rb

```diff
@@ -29,25 +28,67 @@ def import_tracer_profile(tracer_profile)
       end
     end
 
-    def update_kernel_version(version, release)
-      packages = ::Katello::InstalledPackage.where('name LIKE ?', '%kernel%').where.not(name: 'kernelcare').to_a
+    def update_kernel_version(version, release, old_kernel_version)
       delete_ids = []
-      simple_packages = packages.map do |p|
-        delete_ids << p.id
-        ::Katello::Pulp::SimplePackage.new({
-                                             arch: p.arch,
-                                             epoch: p.epoch,
-                                             version: version,
-                                             release: release,
-                                             name: p.name
-                                           })
+      simple_packages = []
+      old_version, old_release = old_kernel_version.split('-')
+      if operatingsystem.family == 'Redhat'
+        packages = installed_packages.select do |package|
+          package.name.include?('kernel') &&
+            package.name != 'kernelcare' &&
+            package.version == old_version &&
+            old_release.include?(package.release)
+        end
+        return if packages.empty?
+
+        simple_packages = packages.map do |p|
+          delete_ids << p.id
+          ::Katello::Pulp::SimplePackage.new({
+                                               arch: p.arch,
+                                               epoch: p.epoch,
+                                               version: version,
+                                               release: release,
+                                               name: p.name
+                                             })
+        end
+        found = import_package_profile_in_bulk(simple_packages)
+        sync_kernel_associations(found.map(&:id).uniq, delete_ids)
       end
-      found = import_package_profile_in_bulk(simple_packages)
-      sync_kernel_associations(found.map(&:id).uniq, delete_ids)
+      if operatingsystem.family == 'Debian'
+        packages = installed_debs.select do |package|
+          package.name.include?('generic') && package.version.include?("#{old_version}-#{old_release}")
+        end
+        return if packages.empty?
+
+        new_kernels = packages.map do |p|
+          delete_ids << p.id
+          name = if p.name.include?("#{old_version}-#{old_release}")
+                   p.name.sub("#{old_version}-#{old_release}", "#{version}-#{release}")
+                 else
+                   p.name
+                 end
+          ::Katello::InstalledDeb.find_or_create_by(name: name,
+                                                    architecture: p.architecture,
+                                                    version: "#{version}-#{release}")
+        end
+        update_debian_kernel_version(new_kernels.map(&:id), delete_ids)
+      end
+    end
+
+    class ::Host::Managed::Jail < Safemode::Jail
+      allow :installed_debs
```

Comment:
Please create a pull request for Katello to add this line to 
https://github.com/Katello/katello/blob/master/app/models/katello/concerns/host_managed_extensions.rb#L481

Reply:
I will create a PR for Katello, but then I think we should wait until a new version of Katello is released to remove this part of code. So I think we can leave it until a new version is released

Reply:
https://github.com/Katello/katello/pull/10398